Sources/HummingbirdCore/Request/Request.swift

```diff
@@ -60,10 +65,100 @@ public struct Request: Sendable {
         self.body = .init(buffer: byteBuffer)
         return byteBuffer
     }
+
+    public func cancelOnInboundClose<Value: Sendable>(_ process: @escaping @Sendable (Request) async throws -> Value) async throws -> Value {
```

Comment:
This should be part of the (Core)RequestContext instead

Reply:
`Server` needs access to this object and it knows nothing about RequestContexts so not sure how we can do this.